### DISCUSSION TITLE
While creating framework term, not ab le to pass list of term object in request payload

### DISCUSSION THREAD

**Discussion by amitpriyadarshi** [Sunbird-Knowlg/Community#195] [Bugs]
[pallakartheekreddy](https://github.com/pallakartheekreddy)
**Problem:** 
While creating framework term request payload with list of terms is not working and throwing server error.
```
Caused by: java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class java.util.Map (java.util.ArrayList and java.util.Map are in module java.base of loader 'bootstrap')
    at controllers.v3.FrameworkTermController.$anonfun$createFrameworkTerm$1(FrameworkTermController.scala:18)
```

**Taxonomy Service Version:** release-5.7.0_RC7

**Working Payload:**
`{
    "request": {
        "term": {
            "name": "CBSE",
            "code": "cbse"
        }
    }
}`

**Non Working Payload:**
`{
    "request": {
        "term": [
            {
                "name": "CBSE",
                "code": "cbse"
            }
        ]
    }
}`

**RCA:**
As per the code walk through we realised `term` in the payload is expected as `Map`, while in the request payload `term` was getting passed as `List of Map`.
But in old implementation request payload with `term` as `Map` and `List of Map` was getting handled.

cc: @vinukumar-vs

**Comment by pallakartheekreddy** [comment]
@amitpriyadarshi Which release version you are using? 
We have resolved this issue as part of release-6.0.0 Knowlg. Please find below JIRA ticker for the same
https://project-sunbird.atlassian.net/browse/KN-968

**Comment by amitpriyadarshi** [comment]
@pallakartheekreddy we used the work around solution i.e. passing term as object instead of list of objects. It is working fine. Also as you mentioned the fix is in 6.0.0, it will work for us.
